<html>
<!-- ####################################################### -->
<!-- #  This html was produced by the igemwiki generator   # -->
<!-- #  https://github.com/igemuoftATG/generator-igemwiki  # -->
<!-- ####################################################### -->

<!-- repo for this wiki: https://github.com/igem-waterloo/wiki2017 -->

</html>
{{Waterloo/head}}
<html>

<nav class="navbar navbar-toggleable-md navbar-inverse bg-primary fixed-top"><a class="navbar-brand" href="http://2017.igem.org/Team:Waterloo">Waterloo</a><button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
<span class="navbar-toggler-icon"></span>
</button><div class="collapse navbar-collapse" id="navbarCollapse"><ul class="navbar-nav mr-auto">
<li class="nav-item"><a class="nav-link" href="http://2017.igem.org/Team:Waterloo"><span>Home</span></a></li>
</li>
<li class="dropdown nav-item"><a href="#" class="nav-link dropdown-toggle nav-link" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Team<span class="caret"></span></a>
<ul class="dropdown-menu">
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Team"><span>Team</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Collaborations"><span>Collaborations</span></a></li>
</li>
</ul>
<li class="active dropdown nav-item"><a href="#" class="nav-link dropdown-toggle nav-link" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Project<span class="caret"></span></a>
<ul class="dropdown-menu">
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Description"><span>Description</span></a></li>
</li>
<li class="active"><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Design"><span>Design</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Experiments"><span>Experiments</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Proof"><span>Proof</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Demonstrate"><span>Demonstrate</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Results"><span>Results</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Notebook"><span>Notebook</span></a></li>
</li>
</ul>
<li class="nav-item"><a class="nav-link" href="http://2017.igem.org/Team:Waterloo/Parts"><span>Parts</span></a></li>
</li>
<li class="nav-item"><a class="nav-link" href="http://2017.igem.org/Team:Waterloo/Safety"><span>Safety</span></a></li>
</li>
<li class="nav-item"><a class="nav-link" href="http://2017.igem.org/Team:Waterloo/Attributions"><span>Attributions</span></a></li>
</li>
<li class="dropdown nav-item"><a href="#" class="nav-link dropdown-toggle nav-link" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Human Practices<span class="caret"></span></a>
<ul class="dropdown-menu">
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Human_Practices"><span>Human Practices</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/HP-Silver"><span>Silver</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/HP-Gold"><span>Gold</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Integrated_Practices"><span>Integrated Practices</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Engagement"><span>Engagement</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Snake35"><span>Snake35</span></a></li>
</li>
</ul>
<li class="dropdown nav-item"><a href="#" class="nav-link dropdown-toggle nav-link" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Awards<span class="caret"></span></a>
<ul class="dropdown-menu">
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Entrepreneurship"><span>Entrepreneurship</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Hardware"><span>Hardware</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Software"><span>Software</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Measurement"><span>Measurement</span></a></li>
</li>
<li class=""><a class="dropdown-item" href="http://2017.igem.org/Team:Waterloo/Model"><span>Model</span></a></li>
</li>
</ul>
</ul>
</div></nav>
<div class="content">
<div class="row"><div class="col"><div class="content-main"><h1 id="project-design">Project Design</h1>
<h2 id="prions">Prions</h2>
<p>Prions are self-perpetuating proteins that have a soluble, normally functional form and an additional insoluble form known as aggregates or amyloids (Tyedmers et.al, 2010). They are self-propagating conformations that have the ability to transmit phenotypes that are not caused by changes in nucleic acid formations (Tyedmers et.al, 2010). The first prions that were discovered were disease-prions, notably in mammalian diseases including sheep scrapie and human Creutzfeldt- Jacob (Liebman &amp; Chernoff, 2012). These diseases were transmitted by infectious conformational isoforms of proteins, which in turn convert other proteins into the isoform (Liebman &amp; Chernoff, 2012). To date, there are over thirty human diseases associated with the formation of prions (Liebman &amp; Chernoff, 2012).</p>
<p>Currently, there are seven-known prions in Saccharomyces cerevisiae, each with a different function and phenotype (Tyedmers et.al, 2010). Although there are no general sequence similarity among all the seven prions, there is a heightened expressed level of asparagine and glutamine residues in all the prions (Tyedmers et.al, 2010).
<br>
<br></p>
<h4 id="-modularity-of-prions-"><em>Modularity of Prions</em></h4>
<p><br>
New prion domains and functions and reporters found to make synthetic prions- leading to new natural prions (Wickner et.al, 2000). A hybrid prion protein can be formed by fusing different prion domains into the corresponding protein filaments (Wickner et.al, 2000). However, this hybridization is very selective - as it will not combine with all protein filaments (Wickner et.al, 2000).
<br>
<br></p>
<h4 id="-reasoning-for-usage-"><em>Reasoning for Usage</em></h4>
<p><br>
By using the self-perpetuating qualities of the prion domain as a tool, alternative forms of various proteins can be induced. This can occur through the enhancement of an existing function, or transferring a new one. Through this mechanism, in future research and applications, isoforms of proteins can be produced and manipulated for the desired effect or product. Applications of this project can be found here.</p>
<h2 id="the-prion-domain">The Prion Domain</h2>
<p>The prion domain, derived from Sup35 of <em>S.cerevisiae</em>, was employed in the design of this project. The utility of Sup35 is based on its ability to manifest in either an infectious or non-infectious state (Dale et al., 2012). In [PSI-], the non-infectious state, the protein acts as a translational termination factor (Dale et al., 2012). In the [PSI+] state, however, the protein confers prion-like properties (Dale et al., 2012)). [PSI+] is often observed upon overexpression of Sup35, which results in protein aggregation (Bradley &amp; Liebman, 2004). Based on the discovery that only the N-terminal and a small portion of the middle domain are essential to the propagation of the prion state, only the coding sequence of the first 137 amino acids was incorporated into the plasmid design (Bradley &amp; Liebman, 2004). Decreasing the size of the plasmid was expected to aid in plasmid construction and transformation. This prion domain is referred to as “PrD”.  </p>
<h2 id="bimolecular-fluorescence-complementation-bifc-">Bimolecular Fluorescence Complementation (BiFC)</h2>
<p>BiFC is employed as a means to visualize protein interactions within live yeast cells. The method is based upon the principle that fragments of a fluorescent protein do not have the ability to fluoresce without interacting with one another in the form of a complex (Kerppola, 2008). In applying this principle to our project, the objective was to utilize the prion domain as a tool to give rise to functional proteins. The interaction of the N-terminal of YFP fused with PrD (PrD-nYFP) and the C-terminal of YFP fused with PrD (PrD-cYFP) is expected to result in the emission of fluorescence when both split proteins come into close proximity in an aggregate.</p>
<h2 id="fluorescence-resonance-energy-transfer-fret-">Fluorescence Resonance Energy Transfer (FRET)</h2>
<p>FRET represents another method to visualize intracellular interactions. The premise underlying this technique is that the transfer of energy from excited fluorophores to non-excited fluorophores can occur when the two fluorophores are within a distance of 5 nm (Alberts et al., 2015). Labelling molecules of interest with different fluorophores enables the recognition of instances when they are at such close proximity, which typically only occurs when the molecules are interacting (Alberts et al., 2015). In order for the energy transfer to occur, the excitation spectrum of one fluorophore must overlap with the emission spectrum of another (Alberts et al., 2015). In this project, the excitation of cyan fluorescent protein (CFP) at approximately 430 nm will result in emission from yellow fluorescent protein (YFP) at approximately 535 nm if the associated prion domain is successful in increasing the interactive potential of the fluorophores (Alberts et al., 2015). FRET, as a part of this project design, is expected to demonstrate the ability of prion domains to bring together different proteins in an attempt to manipulate cellular processes. </p>
<h2 id="references">References</h2>
<ol>
<li><p>Alberts, B., Johnson, A., Lewis, J., Morgan, D., Raff, M., Roberts, K., &amp; Walter, P. (2015). Ways of Working with Cells. In Molecular Biology of the Cell (6th ed., pp. 543-544). New York, NY: Garland Science.</p>
</li>
<li><p>Bradley, M. E., &amp; Liebman, S. W. (2004). The Sup35 domains required for maintenance of weak, strong or undifferentiated yeast [PSI ] prions. Molecular Microbiology,51(6), 1649-1659. doi:10.1111/j.1365-
2958.2003.03955.x</p>
</li>
<li><p>Dale, J. W., Schantz, M. V., &amp; Plant, N. (2012). Products from Native and Manipulated Cloned Genes. In From Genes to Genomes (3rd ed., p. 205). Hoboken, NJ: John Wiley &amp; Sons, Inc.</p>
</li>
<li><p>Kerppola, T. K. (2008). Bimolecular Fluorescence Complementation (BiFC) Analysis as a Probe of Protein Interactions in Living Cells. Annual Review of Biophysics, (37), 465 - 487. doi:10.1146/annurev.biophys.37.032807.125842.</p>
</li>
<li><p>Liebman, S. W., &amp; Chernoff, Y. O. (2012). Prions in Yeast. Genetics, 191(4), 1041-1072. doi:10.1534/genetics.111.137760</p>
</li>
<li><p>Tyedmers, J., Treusch, S., Dong, J., Mccaffery, J. M., Bevis, B., &amp; Lindquist, S. (2010). Prion induction involves an ancient system for the sequestration of aggregated proteins and heritable changes in prion fragmentation. Proceedings of the National Academy of Sciences, 107(19), 8633-8638. doi:10.1073/pnas.1003895107.</p>
</li>
<li><p>Wickner, R., Taylor, K., Edskes, H., &amp; Maddelein, M. (2000). Prions: Portable prion domains. Current Biology,10(9). doi:10.1016/s0960-9822(00)00460-7.</p>
</li>
</ol>
</div></div></div>
</div>
</html>
{{Waterloo/footer}}
